section11/numberformat_proposed.html

```diff
@@ -607,12 +605,27 @@ <h1>GetNumberFormatPattern ( _numberFormat_, _x_, _exponent_ )</h1>
               1. Assert: _compactDisplay_ is `"long"`.
               1. Let _displayNotation_ be `"compactLong"`.
         1. Let _patterns_ be _patterns_.[[&lt;_displayNotation_&gt;]].
-        1. If _x_ is not *NaN* and _x_ &lt; 0 or _x_ is *-0*, then
-          1. Let _pattern_ be _patterns_.[[negativePattern]].
-        1. Else If _x_ is 0, then
-          1. Let _pattern_ be _patterns_.[[zeroPattern]].
+        1. Let _signDisplay_ be _numberFormat_.[[SignDisplay]].
+        1. If _signDisplay_ is `"never"`, then
+          1. Let _patterns_ be _patterns_.[[zeroPattern]].
+        1. Else If _signDisplay_ is `"auto"`, then
+          1. If _x_ is 0 or _x_ &gt; 0 or _x_ is *NaN*, then
+            1. Let _patterns_ be _patterns_.[[zeroPattern]].
+          1. Else,
+            1. Let _pattern_ be _patterns_.[[negativePattern]].
+        1. Else If _signDisplay_ is `"always"`, then
+          1. If _x_ is 0 or _x_ &gt; 0 or _x_ is *NaN*, then
+            1. Let _patterns_ be _patterns_.[[positivePattern]].
```

Comment:
Does this mean we can end up with `"+NaN"`?

Reply:
Yeah; it's an edge case without any particularly clear correct answer, so I made the spec reflect ICU behavior:

https://github.com/unicode-org/icu/blob/maint/maint-66/icu4c/source/test/intltest/numbertest_api.cpp#L2107

We should add this to the README. #73